graylog2-web-interface/src/pages/EnterprisePage.jsx

```diff
@@ -1,48 +1,153 @@
   displayName: 'EnterprisePage',
-  mixins: [Reflux.connect(NodesStore)],
+  mixins: [Reflux.connect(EnterpriseStore)],
 
-  _isLoading() {
-    return !this.state.nodes;
+  componentDidMount() {
+    EnterpriseActions.getLicenseInfo();
   },
 
-  render() {
-    let orderLink = 'https://www.graylog.org/enterprise';
+  onFreeLicenseFormSubmit(formFields, callback) {
+    EnterpriseActions.requestFreeEnterpriseLicense(formFields)
+      .then(() => callback(true))
+      .catch(() => callback(false));
+  },
+
+  _isLoading() {
+    const { licenseStatus } = this.state;
+    return !licenseStatus;
+  },
 
-    if (this.state.clusterId) {
-      orderLink = `https://www.graylog.org/enterprise?cid=${this.state.clusterId}&nodes=${this.state.nodeCount}`;
+  renderLicenseFormContent(licenseStatus) {
+    let licenseFormContent;
+    if (this._isLoading()) {
+      licenseFormContent = <Spinner text="Loading license status" />;
+    } else if (licenseStatus === 'installed') {
+      licenseFormContent = (
+        <Alert bsStyle="success">
+          You have a Graylog Enterprise license installed.
+        </Alert>
+      );
+    } else if (licenseStatus === 'staged') {
+      licenseFormContent = (
+        <Alert bsStyle="warning">
+          You requested a free Graylog Enterprise license. It will be activated once you restart the Graylog
+          server with the Graylog Enterprise plugins installed.
+        </Alert>
+      );
+    } else {
+      licenseFormContent = <EnterpriseFreeLicenseForm onSubmit={this.onFreeLicenseFormSubmit} />;
     }
 
+    return licenseFormContent;
+  },
+
+  render() {
+    const { licenseStatus } = this.state;
+
     return (
-      <DocumentTitle title="Graylog Enterprise">
+      <DocumentTitle title="Try Graylog Enterprise">
         <div>
-          <PageHeader title="Graylog Enterprise">
+          <PageHeader title="Try Graylog Enterprise">
             {null}
 
             <span>
               Graylog Enterprise adds commercial functionality to the Open Source Graylog core. You can learn more
-              about Graylog Enterprise and order a license on the <a href={orderLink} target="_blank">product page</a>.
-            </span>
-
-            <span>
-              <a className="btn btn-lg btn-success" href={orderLink} target="_blank">Order a license</a>
+              about Graylog Enterprise on the <EnterpriseProductLink>product page</EnterpriseProductLink>.
             </span>
           </PageHeader>
 
-          <GraylogClusterOverview />
-          <PluginList />
+          <GraylogClusterOverview layout="compact">
+            <PluginList />
+          </GraylogClusterOverview>
+          <IfPermitted permissions="freelicenses:create">
+            <Row className="content">
+              <Col md={6}>
+                <BiggerFontSize>
+                  <p>Extend Graylog’s Open Source capabilities with a free trial of Graylog Enterprise (up to 5 GB/Day).
```

Comment:
I would add a break after the first sentence to make it stand out a little more.